docs/zero-to-hero/02-beginner/05-logging-in.md

```diff
@@ -0,0 +1,168 @@
+
+So the key is simply allowed to use the allowance in NEAR on gas, deducting from the account for each function call.
+
+#### Second key:
+
+```js
+{
+  access_key: {
+    nonce: 72777733000000,
+    permission: {
+      FunctionCall: {
+        allowance: '250000000000000000000000', // 0.25 NEAR, which is a typical allowance
+        method_names: [ 'foo', 'bar' ], // Can call methods foo and bar only
+        receiver_id: 'puzzle.testnet'
+      }
+    }
+  },
+  public_key: 'ed25519:CM4JtNo2sL3qPjWFn4MwusMQoZbHUSWaPGCCMrudZdDU'
+},
+```
+
+This second key specifies which methods can be called, and has a lower allowance.
+
+Note that the allowance for this key (a quarter of a NEAR) is the default allowance when a person "logs in" in with the NEAR Wallet. 
+
+In NEAR, "logging in" typically means adding a key like this to your account. We'll cover this more in a moment.
+
+#### Third key
+
+```js
+{
+  access_key: { nonce: 72770704000019, permission: 'FullAccess' },
+  public_key: 'ed25519:FG4HjEPsvP5beScC3hkTLztQH8k9Qz9maTaumvPDa5t3'
+}
+```
+
+The third key is a full-access key.
+
+Since this key can perform all the Actions, there aren't additional details or restrictions like the function-call access keys we saw.
+
+## What does "log in" mean in a blockchain context?
+
+Let's take a step back from NEAR and talk about how login works broadly using web3 wallets.
+
+A web3 wallet (like Ethereum's MetaMask, Cosmos's Keplr, or the NEAR Wallet) stores a private key for an account. When interacting with decentralized apps, a user will typically use the wallet to sign transactions and send them to the blockchain for processing.
+
+However, web3 wallets can also be used to sign any kind of message, and it doesn't need to send anything to the blockchain. This is sometimes called "offline signing" and protocols will sometimes create standards around how to sign data.
+
+In other ecosystems, the idea of "logging in" with a web3 wallet uses this offline signing. A user is asked to sign a structured message and a backend can confirm that the message was signed by a given account.
+
+NEAR keys can also sign and verify messages in this manner. In fact, there are a couple simple examples of how to achieve this in the [`near-api-js` cookbook](https://github.com/near/near-api-js/blob/master/examples/cookbook/utils/verify-signature.js).
+
+There are potential drawbacks to this offline signing technique. For instance, if you wish to log out of an application (or retract your participation) you must put trust in someone else's backend. Or worse, what if a malicious person intercepts your signed message, and you wish to kick them out? We quickly see that using a web3 wallet for signed typed data runs into limitations.
```

Comment:
I don't know what you mean by put trust in someone else's backend. Do you mean to make sure they don't lose the signed message?

Reply:
Thanks, Austin, I've tried to clarify here:
https://github.com/near/sdk-docs/pull/72/commits/6ed22244a10c2bd90ec29fd74302b5dbb53f4c41